### PR TITLE
OAK-9339 - Image Similarity: LSH based search

### PR DESCRIPTION
**Pull request by averma21**

### CODE REVIEW

File: oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticPropertyDefinition.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        private final String queryTimeSimilarityFunction;
+        /**
+         * Possible values l2 (with lsh or exact model), l1 (with exact model), A (angular distance - with exact model)
+         */
+        private final String indexTimeSimilarityFunction;
+        /**
+         * Take the top vectors with the most matching hashes and compute their exact similarity to the query vector. The candidates parameter
+         * controls the number of exact similarity computations. Specifically, we compute exact similarity for the top candidates candidate vectors
+         * in each segment. As a reminder, each Elasticsearch index has >= 1 shards, and each shard has >= 1 segments. That means if you set
+         * "candiates": 200 for an index with 2 shards, each with 3 segments, then you’ll compute the exact similarity for 2 * 3 * 200 = 1200 vectors.
```
Comment (fabriziofortino) [inline oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticPropertyDefinition.java]: typo
```suggestion
         * "candidates": 200 for an index with 2 shards, each with 3 segments, then you’ll compute the exact similarity for 2 * 3 * 200 = 1200 vectors.
```

File: oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexHelper.java [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
 import org.apache.jackrabbit.oak.api.Type;
 import org.apache.jackrabbit.oak.plugins.index.elastic.ElasticIndexDefinition;
+import org.apache.jackrabbit.oak.plugins.index.elastic.ElasticPropertyDefinition;
```
Comment (fabriziofortino) [inline oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexHelper.java:21]: `ES_DENSE_VECTOR_TYPE` at line 38 is never used. Can we remove it?

**Review by fabriziofortino — APPROVED** [review]
LGTM

**Review by thomasmueller — APPROVED** [review]

File: oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexHelper.java (hunk 2 above)
Comment (averma21) [reply oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexHelper.java:21]: Done.

**Comment by averma21** [comment]
Changes committed in http://svn.apache.org/viewvc?view=revision&revision=1886318

**Pull request closed without merging**